src/plumbing/map.cljx

```diff
@@ -14,7 +14,8 @@
   "Like select-keys, but asserts that all keys are present."
   [m ks]
   (doseq [k ks]
-    (assert (contains? m k)))
+    (or (contains? m k)
+        (schema/assert-iae false "Key %s not found in %s" k (mapv key m))))
```

Comment:
Finally, I've been meaning to modify `safe-get` to ellipsize the keys, since the error message is not particularly nice when it fails on a map with millions of entries.  Not necessary to do that as part of this change, but bonus points if you're feeling up to it :).